src/user-guide/writing-workflows/runtime.rst

```diff
@@ -634,18 +637,40 @@ can be:
 Values should be a list of commands, command lines, or command line templates
 (see below) to call if the specified event is triggered.
 
+
+.. _user_guide.runtime.task_event_handling.general_event_handlers:
+
 General Event Handlers
 ^^^^^^^^^^^^^^^^^^^^^^
 
-Alternatively you can configure ``handler events`` and ``handlers``
-under :cylc:conf:`[runtime][<namespace>][events]`, where the former is a list
-of events (as above), and the latter a list of commands, command lines, lines
-or command line templates (see below) to call if any of the specified events
-are triggered.
+.. cylc-scope:: flow.cylc[runtime][<namespace>][events]
+
+Alternatively you can configure a list of generic event :cylc:conf:`handlers` to be run
+for configured :cylc:conf:`handler events`.
+
+:cylc:conf:`handler events`
+   A list of events which may include any of the above
+   events (e.g. ``submission failed`` or ``warning``) or
+   any of a task's :term:`custom outputs <custom output>`.
```

Comment:
This is piece of information which was missing before.

Turns out custom outputs can be used in `handler events`, this is covered by an existing functional test.